java/vector/src/test/java/org/apache/arrow/vector/TestCopyFrom.java

```diff
@@ -1069,4 +1070,29 @@ public void testCopyFromWithNulls14() {
       }
     }
   }
+
+  @Test //https://issues.apache.org/jira/browse/ARROW-7837
+  public void testCopySafeArrow7837() {
+    // this test exposes a bug in `handleSafe` where
+    // it reads a stale index and as a result missed a required resize of the value vector.
+    try (VarCharVector vc1 = new VarCharVector("vc1", allocator);
+         VarCharVector vc2 = new VarCharVector("vc2", allocator);
+    ) {
+      vc2.setInitialCapacity(20, 0.5);
```

Comment:
are the exact values important?  if so please add a comment.

also, please try add parameter name comments to constants i.e. (/*valueCount=*/20, /*density=*/0.5)